qiskit/_qiskiterror.py

```diff
@@ -23,6 +23,6 @@ def __str__(self):
         return repr(self.message)
 
 
-class QISKitIndexError(QISKitError, IndexError):
+class QiskitIndexError(QiskitError, IndexError):
```

Comment:
Same here as above, although to a slightly lesser degree because this isn't the base exception.

Reply:
This probably isn't needed since we don't expose this externally and only raised in _register